Python/errors.c

```diff
@@ -1603,6 +1603,19 @@ PyErr_ProgramTextObject(PyObject *filename, int lineno)
     return err_programtext(tstate, fp, lineno);
 }
 
+void _Py_NO_RETURN
+Py_UNREACHABLE(void)
+{
+#ifdef Py_DEBUG
+    Py_FatalError(
+        "We've reached an unreachable state. Anything is possible.\n"
+        "The limits were in our heads all along. Follow your dreams.\n"
+        "https://xkcd.com/2200");
+#else
+    Py_FatalError("Python internals reached an unreachable code path");
```

Comment:
"Python internals" isn't quite accurate; since this is a public function it may be used in third-party code (which is probably more likely to actually hit this function :)), but "Python internals" makes it sound like it's definitely our bug and so we'll get the report.  Maybe something like "Unreachable C code path reached"?

Reply:
I changed the message.